src/Zicht/Bundle/UrlBundle/Url/DelegatingProvider.php

```diff
@@ -37,15 +37,18 @@ public function __construct()
      */
     public function addProvider(Provider $provider, $priority = 0)
     {
-        $this->providers->insert($provider, $priority);
+        $this->providers[] = [
+            'priority' => $priority,
+            'provider' => $provider
+        ];
```

Comment:
Why not sort when we add a provider?  I.e. add `$this->providers = iterable($this->providers)->sorted('priority')->values()`, this should provide a performance benefit.

Reply:
Is this a performance benefit when you add, lets say, 20 providers and don't use this class at all? Providers are added runtime with every request I assume